src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs

```diff
@@ -344,6 +353,16 @@ private bool IsPluginPossiblyAvailable()
             return !string.IsNullOrEmpty(_rawPluginPaths);
         }
 
+        private void OnPluginClosed(object sender, EventArgs e)
+        {
+            if (sender is IPlugin plugin)
+            {
+                plugin.Closed -= OnPluginClosed;
+
+                _pluginUtilities.TryRemove(plugin.Id, out Lazy<IPluginMulticlientUtilities> utilities);
```

Comment:
How did you catch this? :) 
Did 1 client process ever try to spawn the same plugin exe twice?

Reply:
No, while reusing `_pluginUtilities` to ensure one-time (per plugin instance) serialization of the cache file I noticed that old entries for exited plugin instances were never purged.  It was a small memory leak that I fixed at the same time.